### PR TITLE
delete gateway in cleanup-gateway-resouces

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Changes proposed in this PR ###  
- Delete meshgateways on resource cleanup if they exist

### How I've tested this PR ###
- Local helm install/helm uninstall and checked logs of resource cleanup pod

### How I expect reviewers to test this PR ###
- local helm install/uninstall
- glance over

### Checklist ###
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: charts/consul/templates/gateway-resources-configmap.yaml [hunk 1]
```diff
@@ -69,7 +69,6 @@ data:
             {{- with .Values.meshGateway.dnsPolicy }}
             dnsPolicy: {{ . }}
             {{- end }}
-            priorityClassName: {{ toJson .Values.meshGateway.priorityClassName }}
```
Comment (sarahalsmiller) [inline charts/consul/templates/gateway-resources-configmap.yaml:72]: Duplicated line

File: control-plane/gateways/config.go [hunk 2]
```diff
@@ -38,3 +41,8 @@ type GatewayConfig struct {
 	// defined on a Gateway.
 	MapPrivilegedServicePorts int
 }
+
+type GatewayResources struct {
```
Comment (sarahalsmiller) [inline control-plane/gateways/config.go:45]: Both gateway resources and gateway cleanup need this struct so I pulled it into the gateways package for ease of use

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/hashicorp/consul-k8s/control-plane/gateways"
```
Comment (missylbytes) [inline control-plane/subcommand/gateway-cleanup/command.go:11]: nit: Can you move this import

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 4]
```diff
@@ -277,13 +281,13 @@ func (c *Command) loadGatewayConfigs() error {
 	return nil
 }
 
-func (c *Command) deleteV2GatewayClassAndClassConfigs() error {
+func (c *Command) deleteV2GatewayClassAndClassConfigs(ctx context.Context) error {
```
Comment (jm96441n) [inline control-plane/subcommand/gateway-cleanup/command.go:284]: nice catch!

File: control-plane/gateways/config.go (hunk 2 above)
Comment (jm96441n) [reply control-plane/gateways/config.go:45]: good call, I like this change

**Review by missylbytes — APPROVED** [review]
LGTM besides the nit

**Review by NiniOak — APPROVED** [review]
Looks good!

**Pull request merged**